### PR TITLE
[BISERVER-13699] Add functionality for user to reset password

### PR DESCRIPTION
**Pull request by angel-ramoscardona**

### CODE REVIEW

**Comment by buildguy** [comment]
### :white_check_mark: Build finished in 32m 18s
Build command:
```console
mvn clean verify -B -e -Daudit -amd -pl extensions,user-console
```
---
:ok_hand: All tests passed!
 
Tests run: **1498**, Failures: **0**, Skipped: **1** &nbsp;&nbsp; [Test Results](http://wingman.pentaho.net/job/projects/job/platform/job/PR-4681/1/testReport)

 ---
 
 

:information_source:  _This is an automatic message_

**Comment by buildguy** [comment]
### :white_check_mark: Build finished in 32m 6s
Build command:
```console
mvn clean verify -B -e -Daudit -amd -pl extensions,user-console
```
---
:ok_hand: All tests passed!
 
Tests run: **1498**, Failures: **0**, Skipped: **1** &nbsp;&nbsp; [Test Results](http://wingman.pentaho.net/job/projects/job/platform/job/PR-4681/2/testReport)

 ---
 
 

:information_source:  _This is an automatic message_

**Comment by buildguy** [comment]
### :white_check_mark: Build finished in 33m 52s
Build command:
```console
mvn clean verify -B -e -Daudit -amd -pl extensions,user-console
```
---
:ok_hand: All tests passed!
 
Tests run: **1498**, Failures: **0**, Skipped: **1** &nbsp;&nbsp; [Test Results](http://wingman.pentaho.net/job/projects/job/platform/job/PR-4681/3/testReport)

 ---
 
 

:information_source:  _This is an automatic message_

**Comment by buildguy** [comment]
### :white_check_mark: Build finished in 33m 53s
Build command:
```console
mvn clean verify -B -e -Daudit -amd -pl extensions,user-console
```
---
:ok_hand: All tests passed!
 
Tests run: **1498**, Failures: **0**, Skipped: **1** &nbsp;&nbsp; [Test Results](http://wingman.pentaho.net/job/projects/job/platform/job/PR-4681/4/testReport)

 ---
 
 

:information_source:  _This is an automatic message_

File: user-console/src/main/java/org/pentaho/mantle/client/admin/ChangePasswordByUserDialog.java [hunk 1]
```diff
@@ -0,0 +1,165 @@
...
+import com.google.gwt.user.client.ui.Panel;
+import com.google.gwt.user.client.ui.PasswordTextBox;
+import com.google.gwt.user.client.ui.SimplePanel;
+import com.google.gwt.user.client.ui.VerticalPanel;
+import org.pentaho.gwt.widgets.client.utils.string.StringUtils;
+import org.pentaho.mantle.client.messages.Messages;
+import org.pentaho.mantle.client.ui.xul.MantleController;
+import org.pentaho.ui.xul.gwt.tags.GwtDialog;
+
+@SuppressWarnings( "squid:S110" )
```
Comment (peterrinehart) [inline user-console/src/main/java/org/pentaho/mantle/client/admin/ChangePasswordByUserDialog.java:39]: Can we get a comment here about which sonar rule we're suppressing and why?

**Comment by buildguy** [comment]
### :white_check_mark: Build finished in 33m 50s
Build command:
```console
mvn clean verify -B -e -Daudit -amd -pl extensions,user-console
```
---
:ok_hand: All tests passed!
 
Tests run: **1498**, Failures: **0**, Skipped: **1** &nbsp;&nbsp; [Test Results](http://wingman.pentaho.net/job/projects/job/platform/job/PR-4681/5/testReport)

 ---
 
 

:information_source:  _This is an automatic message_

**Comment by buildguy** [comment]
### :white_check_mark: Build finished in 32m 33s
Build command:
```console
mvn clean verify -B -e -Daudit -amd -pl extensions,user-console
```
---
:ok_hand: All tests passed!
 
Tests run: **1498**, Failures: **0**, Skipped: **1** &nbsp;&nbsp; [Test Results](http://wingman.pentaho.net/job/projects/job/platform/job/PR-4681/6/testReport)

 ---
 
 

:information_source:  _This is an automatic message_

File: extensions/src/main/java/org/pentaho/platform/web/http/api/resources/services/UserRoleDaoService.java [hunk 2]
```diff
@@ -21,6 +21,7 @@
 package org.pentaho.platform.web.http.api.resources.services;
```
Comment (NJtwentyone) [inline extensions/src/main/java/org/pentaho/platform/web/http/api/resources/services/UserRoleDaoService.java:21]: update copyright header date

File: extensions/src/main/java/org/pentaho/platform/web/http/api/resources/UserRoleDaoResource.java [hunk 3]
```diff
@@ -69,6 +69,7 @@
 @Path ( "/userroledao/" )
 public class UserRoleDaoResource extends AbstractJaxRSResource {
```
Comment (NJtwentyone) [inline extensions/src/main/java/org/pentaho/platform/web/http/api/resources/UserRoleDaoResource.java:70]: update copyright header date

File: user-console/src/main/java/org/pentaho/mantle/client/admin/ChangePasswordByUserDialog.java (hunk 1 above)
Comment (NJtwentyone) [reply user-console/src/main/java/org/pentaho/mantle/client/admin/ChangePasswordByUserDialog.java:40]: author resolved the comment with comment

File: user-console/src/main/java/org/pentaho/mantle/client/admin/ChangePasswordByUserDialog.java [hunk 4]
```diff
@@ -0,0 +1,166 @@
...
+ * This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY;
+ * without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+ * See the GNU Lesser General Public License for more details.
+ *
+ *
+ * Copyright (c) 2002-2020 Hitachi Vantara. All rights reserved.
+ *
+ */
+
+package org.pentaho.mantle.client.admin;
```
Comment (NJtwentyone) [inline user-console/src/main/java/org/pentaho/mantle/client/admin/ChangePasswordByUserDialog.java:21]: I don't think it is valid, getting code smell from sonar "1 duplicated blocks of code must be removed."

File: user-console/src/main/java/org/pentaho/mantle/client/admin/UserRolesAdminPanelController.java [hunk 5]
```diff
@@ -51,6 +51,8 @@
 
 public class UserRolesAdminPanelController extends UserRolesAdminPanel implements ISysAdminPanel,
```
Comment (NJtwentyone) [inline user-console/src/main/java/org/pentaho/mantle/client/admin/UserRolesAdminPanelController.java:52]: update copyright header date

File: user-console/src/main/java/org/pentaho/mantle/client/ui/xul/MantleModel.java [hunk 6]
```diff
@@ -22,6 +22,7 @@
 
 import org.pentaho.gwt.widgets.client.toolbar.ToolbarButton;
 import org.pentaho.mantle.client.MantleApplication;
+import org.pentaho.mantle.client.admin.ChangePasswordByUserDialog;
```
Comment (NJtwentyone) [inline user-console/src/main/java/org/pentaho/mantle/client/ui/xul/MantleModel.java:25]: update copyright header date

File: user-console/src/main/java/org/pentaho/mantle/client/ui/xul/MantleController.java [hunk 7]
```diff
@@ -47,6 +47,7 @@
 import org.pentaho.mantle.client.admin.EmailAdminPanelController;
 import org.pentaho.mantle.client.admin.ISysAdminPanel;
 import org.pentaho.mantle.client.admin.JsSysAdminPanel;
```
Comment (NJtwentyone) [inline user-console/src/main/java/org/pentaho/mantle/client/ui/xul/MantleController.java:49]: update copyright header date

**Comment by buildguy** [comment]
### :white_check_mark: Build finished in 33m 55s
Build command:
```console
mvn clean verify -B -e -Daudit -amd -pl extensions,user-console
```
---
:ok_hand: All tests passed!
 
Tests run: **1498**, Failures: **0**, Skipped: **1** &nbsp;&nbsp; [Test Results](http://wingman.pentaho.net/job/projects/job/platform/job/PR-4681/7/testReport)

 ---
 
 

:information_source:  _This is an automatic message_

**Pull request merged**